### ISSUE TITLE
[2.0.x] Inconsistent extrusion

### ISSUE THREAD

**Issue by ktand** [MarlinFirmware/Marlin#11822]
I have an issue which seems to be caused by inconsistent extrusion. I'm running a STM32F4 based board in a Prusa clone and a RAMPS board in another Prusa clone. Both printers are using the same stepper drivers and the same velocity and acceleration parameters. `LIN_ADVANCE `is disabled but both printers are using `ADAPTIVE_STEP_SMOOTHING `and `S_CURVE_ACCELERATION`. The STM32F4 printer is running the latest bugfix-2.0.x code while the RAMPS printer is running 1.9. Prints from the RAMPS printer looks MUCH nicer.

It seems like there's some inconsistent extrusion on the STM32F4 machine. Every printed segment is smaller in the middle and wider in the ends. The width of the object is not consistent, it varies between 1.8 and 1.94mm. The "corners" are much more jaggy on the STM32F4 machine. My guess is that it overextrudes at the beginning and end of each segment.

I have attached the G-Code that was used on both machines.

Any ideas what could be causing this?

I did reduce the flow rate to make the problem more visible in the pictures.

![p9135867](https://user-images.githubusercontent.com/17763620/45506268-06f92e00-b78f-11e8-83cf-33b27405890a.JPG)

### Pictures of prints using the STM32F4 machine

![p9135848](https://user-images.githubusercontent.com/17763620/45506277-0bbde200-b78f-11e8-948d-1f2370c2adc0.JPG)
![p9135854](https://user-images.githubusercontent.com/17763620/45506279-0bbde200-b78f-11e8-8464-22eb4bde897b.JPG)
![p9135859](https://user-images.githubusercontent.com/17763620/45506280-0bbde200-b78f-11e8-82dc-34ef830255e0.JPG)

![2018-09-13 20_11_29-window](https://user-images.githubusercontent.com/17763620/45507050-4e80b980-b791-11e8-8583-2792257f5da5.png)

[Calibration Curve Problem.zip](https://github.com/MarlinFirmware/Marlin/files/2380480/Calibration.Curve.Problem.zip)

**Comment by ModMike** [comment]
I too am struggling with inconsistent extrusion on my Azteeg X5 Mini Wifi.

Similar issues as above and crazy ABS multiplier is at .84 just to get right wall thickness.  I also have LIN_ADVANCE disabled and S_CURVE_ACCELERATION enabled.

I tried with and without S-Curve, same issue.  I've redone the config.h twice from scratch.

I also noticed that the retract movement seems to be more than asked for.
